ssl/statem/statem_lib.c

```diff
@@ -1007,6 +999,34 @@ int ssl_choose_client_version(SSL *s, int version)
 }
 
 /*
+ * ssl_verify_client_session_version - Verify that provided SSL_SESSION matches
+ * the protocol negotiated with server.
+ *
+ * @s: client SSL handle.
+ *
+ * Returns 1 on success, 0 on failure
+ */
+int ssl_verify_client_session_version(SSL *s)
+{
+  int err;
+
+  if (s->session == NULL) {
+      return 1;
+  }
+
+  /* Empty session and no EAP-FAST */
+  if (s->session->session_id_length == 0 && !s->session->tlsext_tick) {
```

Comment:
As far as I understood it `tlsext_tick` is non-`NULL` when we are doing `EAP-FAST`, and `session_id` is empty. This condition attempts to catch empty sessions that we just created by the state machine itself.